go-controller/pkg/ovn/ovn.go

```diff
@@ -1116,3 +1116,9 @@ func (oc *Controller) StartServiceController(wg *sync.WaitGroup, runRepair bool)
 	}()
 	return nil
 }
+
+func (oc *Controller) VisitNsInfo(ns string, visitor DoWithNsInfo) error {
+	nsInfo := oc.ensureNamespaceLocked(ns)
```

Comment:
A good pattern here is to log a stack trace if the visitor takes more than N seconds.

This should be gated behind a rate-limiter.

Reply:
Absolutely. Logging a strack trace is a great idea. Although I am not sure why you would rate limit? I mean what is the benefit of doing so - it is work that still needs to happen. Do you mean like allow only M number of simultaneous calls to visit the `nsInfo`?

Reply:
It would be to avoid flooding the logs on a system where nbdb slows down for some reason. In other words, don't log more than N stack traces per second, even if they would other wise qualify.

Reply:
aah, you mean rate limited logging. yes, that makes more sense.